dohproxy/client_protocol.py

```diff
@@ -30,18 +29,10 @@ def __init__(self, args, logger=None):
     async def setup_client(self):
         # Open client connection
         self.logger.debug('Opening connection to {}'.format(self.args.domain))
-        if self.args.insecure:
-            sslctx = ssl.SSLContext(ssl.PROTOCOL_SSLv23)
-            sslctx.options |= ssl.OP_NO_SSLv2
-            sslctx.options |= ssl.OP_NO_SSLv3
-            sslctx.options |= ssl.OP_NO_COMPRESSION
-            sslctx.set_default_verify_paths()
-        else:
-            sslctx = ssl.create_default_context()
-
-        sslctx.set_alpn_protocols(constants.DOH_H2_NPN_PROTOCOLS)
-        sslctx.set_npn_protocols(constants.DOH_H2_NPN_PROTOCOLS)
-
+        sslctx = utils.create_custom_ssl_context(
+            self.args.insecure,
+            self.args.cafile
```

Comment:
for the sake of being explicit.... let's use keyword argument rather than positional.